logsmanagement/flb_plugin.c

```diff
@@ -0,0 +1,982 @@
+        if(likely(syslog_severity)){
+            snprintf(syslog_severity_s, 2, "%.*s", (int) syslog_severity_size, syslog_severity);
+            if(likely(str2int(&syslog_severity_d, syslog_severity_s, 10) == STR2XX_SUCCESS)){
+                p_file_info->flb_tmp_systemd_metrics.sever[syslog_severity_d]++;
+            } // else parsing errors ++ ??
+        } else p_file_info->flb_tmp_systemd_metrics.sever[8]++; // 'unknown'
+
+        /* Parse syslog_facility char* field into int and extract metrics. 
+         * syslog_facility_s will consist of up to 2 chars (plus '\0'), 
+         * see https://datatracker.ietf.org/doc/html/rfc5424#section-6.2.1 */
+        if(likely(syslog_facility)){
+            snprintf(syslog_facility_s, 3, "%.*s", (int) syslog_facility_size, syslog_facility);
+            if(likely(str2int(&syslog_facility_d, syslog_facility_s, 10) == STR2XX_SUCCESS)){
+                p_file_info->flb_tmp_systemd_metrics.facil[syslog_facility_d]++;
+            } // else parsing errors ++ ??
+        } else p_file_info->flb_tmp_systemd_metrics.facil[24]++; // 'unknown'
+
+        if(likely(syslog_severity && syslog_facility)){
+            /* Definition of syslog priority value == facility * 8 + severity */
+            syslog_prival_d = syslog_facility_d * 8 + syslog_severity_d; 
+            syslog_prival_s_size = snprintf(syslog_prival_s, 4, "%d", syslog_prival_d);
+            m_assert(syslog_prival_s_size < 4 && syslog_prival_s_size > 0, "error with snprintf()");
+    
+            new_tmp_text_size += syslog_prival_s_size + 2; // +2 for '<' and '>'
+
+            p_file_info->flb_tmp_systemd_metrics.prior[syslog_prival_d]++;
+        } else {
+            new_tmp_text_size += 3; // +3 for "<->" string
+            p_file_info->flb_tmp_systemd_metrics.prior[192]++; // 'unknown'
+        } 
+
+        char syslog_time_from_flb_time[25]; // 25 just to be on the safe side, but 16 + 1 chars bytes needed only.
+        if(unlikely(!syslog_timestamp)){
+            const time_t ts = tmp_time.tm.tv_sec;
+            struct tm *const tm = localtime(&ts);
+
+            strftime(syslog_time_from_flb_time, sizeof(syslog_time_from_flb_time), "%b %d %H:%M:%S ", tm);
+            new_tmp_text_size += SYSLOG_TIMESTAMP_SIZE;
+        }
+
+        if(unlikely(!syslog_identifier)) new_tmp_text_size += sizeof(UNKNOWN) - 1;
+        if(unlikely(!pid)) new_tmp_text_size += sizeof(UNKNOWN) - 1;
+
+        new_tmp_text_size += 5; // +5 for '[', ']', ':' and ' ' characters around and after pid and '\n' at the end
+
+        /* Metrics extracted, now prepare circular buffer for write */
+        const size_t available_text_space = circ_buffer_prepare_write(buff, new_tmp_text_size);
+        m_assert(available_text_space != 0, "available_text_space is 0");
+        // TODO: What do if available_text_space == 0 ??
+
+        size_t tmp_item_off = buff->in->text_size;
+
+        memcpy(&buff->in->data[tmp_item_off++], "<", 1);
```

Comment:
I suggest to assign a value direct for the vector instead to call a function to do this. It is less expensive.

Reply:
This suggestion is applied for next calls for `memcpy`.

Reply:
Really good point, fixed!